### PR TITLE
[Editor] Remove event listeners with `AbortSignal.any()`

### PR DESCRIPTION
**Pull request by Snuffleupagus**
There's a fair number of event listeners in the editor-code that we're currently removing "manually", by keeping references to their event handler functions.
This was necessary since we have a "global" `AbortController` that applies to all event listeners used in the editor-code, however it's now possible to combine multiple `AbortSignal`s; please see https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/any_static

Since this functionality is [fairly new](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/any_static#browser_compatibility) the viewer will check that `AbortSignal.any()` is available before enabling the editing-functionality.
(It should hopefully be fairly straightforward, famous last words, for users to implement a polyfill to allow editing in older browsers.)

Finally, this patch also adds checks and test-only asserts to ensure that we don't add duplicate event listeners in various editor-code.

### CODE REVIEW

File: web/viewer.html [hunk 1]
```diff
@@ -623,7 +623,7 @@
                     </div>
                   </div>
                   <button id="deleteModelButton" type="button" class="secondaryButton" tabindex="0"><span data-l10n-id="pdfjs-editor-alt-text-settings-delete-model-button">Delete</span></button>
-                  <button id="downloadModelButton"type="button" class="secondaryButton" tabindex="0"><span data-l10n-id="pdfjs-editor-alt-text-settings-download-model-button">Download</span></button>
+                  <button id="downloadModelButton" type="button" class="secondaryButton" tabindex="0"><span data-l10n-id="pdfjs-editor-alt-text-settings-download-model-button">Download</span></button>
```
Comment (Snuffleupagus) [inline web/viewer.html:626]: This is obviously unrelated, but I couldn't be bothered to submit a separate patch just to add the missing space here :-)

**Review by calixteman — APPROVED** [review]
LGTM. Thank you for doing that.
In reading the docs for `AbortSignal::any` I noticed `AbortSignal::Timeout`, is it something we could use somewhere ?

**Comment by Snuffleupagus** [comment]
> In reading the docs for `AbortSignal::any` I noticed `AbortSignal::Timeout`, is it something we could use somewhere ?

I'm sure that could be helpful, but off the top of my head I don't know of any cases that would benefit from it.
